lib/ReactGridLayout.jsx

```diff
@@ -250,10 +250,18 @@ export default class ReactGridLayout extends React.Component {
 
     // Set state
     this.setState({
-      layout: compact(layout, this.props.verticalCompact),
       activeDrag: null,
       oldDragItem: null
     });
+
+    // Set State and callback, but only if layout changed
+    const newLayout = compact(layout, this.props.verticalCompact);
+    if (!isEqual(newLayout, this.state.layout)) {
```

Comment:
This actually won't fire because the compact/move functions modify the layout, they don't clone it.

Perhaps they should - I'll have to see how the perf stacks up.